### PR TITLE
Improve join form validation & make phone number optional

### PR DESCRIPTION
**Pull request by Andrew-Dickinson**
Adds much more user-friendly form validation. Instead of just disabling the submit button with no explanation, the button is enabled, but if they click it too early, they get red text by the field that is required

Also makes phone number optional, consistent with the existing Google Form

https://github.com/user-attachments/assets/bf3c9b5e-8459-4799-baa5-d657ad2775ed

### CODE REVIEW

File: tests/mock/handlers.ts [hunk 1]
```diff
@@ -84,6 +88,12 @@ export default [
         return HttpResponse.json(r, { status: 409 });
       }
 
+      if (joinRequest.phone_number === "") {
+        // This is probably the wrong way to do this, but the empty phone number case was
+        // failing the strict equality check below
+        return HttpResponse.json(good_response, { status: 201 });
+      }
```
Comment (Andrew-Dickinson) [inline tests/mock/handlers.ts:95]: 🫣🫣

**Comment by WillNilges** [comment]
Fucking banger, I'll review this in depth tomorrow after a sleep.

**Review by WillNilges — COMMENTED** [review]
All LGTM, but I suggest we provide translations before we merge.

File: messages/en.json
[the remark this answers is not in the document]
Comment (WillNilges) [reply messages/en.json:1]: https://github.com/nycmeshnet/meshforms/pull/136/files

**Review by WillNilges — APPROVED** [review]

File: tests/mock/handlers.ts (hunk 1 above)
Comment (WillNilges) [reply tests/mock/handlers.ts:95]: oh my lord

Comment (WillNilges) [reply tests/mock/handlers.ts:95]: You could make another set of sample data and deepequals on that?

**Review by WillNilges — CHANGES_REQUESTED** [review]
Could you try my suggestion before merging this?

File: tests/mock/handlers.ts (hunk 1 above)
Comment (WillNilges) [reply tests/mock/handlers.ts:95]:
```suggestion
      if (joinRequest.phone_number === "") {
        // This is probably the wrong way to do this, but the empty phone number case was
        // failing the strict equality check below
        
        let expectedData: JoinFormValues = Object.assign({}, expectedAPIRequestData);
        expectedData.phone_number = "";
        if (isDeepStrictEqual(joinRequest, expectedData)) {
          return HttpResponse.json(good_response, { status: 201 });
        }
      }
```

**Review by WillNilges — APPROVED** [review]

**Pull request merged**